substrate/frame/message-queue/src/tests.rs

```diff
@@ -1662,3 +1662,174 @@ fn integrity_test_checks_service_weight() {
 		}
 	});
 }
+
+/// Test for <https://github.com/paritytech/polkadot-sdk/issues/2319>.
+#[test]
+fn regression_issue_2319() {
+	build_and_execute::<Test>(|| {
+		Callback::set(Box::new(|_, _| {
+			MessageQueue::enqueue_message(mock_helpers::msg("anothermessage"), There);
+		}));
+
+		use MessageOrigin::*;
+		MessageQueue::enqueue_message(msg("callback=0"), Here);
+
+		// while servicing queue Here, "anothermessage" of origin There is enqueued in
+		// "firstmessage"'s process_message
+		assert_eq!(MessageQueue::service_queues(1.into_weight()), 1.into_weight());
+		assert_eq!(MessagesProcessed::take(), vec![(b"callback=0".to_vec(), Here)]);
+
+		assert_eq!(MessageQueue::service_queues(1.into_weight()), 1.into_weight());
+		// expects anothermessage to be processed, but it isn't procssed at all
```

Comment:
But it is processed, right?

Reply:
Yea. It is the regression test from the original issue; i just copy&pasted it. Now since it is fixed it does not fail anymore and the comment is wrong. Will fix 👍